Client/game_sa/CFxSA.cpp

```diff
@@ -11,6 +11,25 @@
 
 #include "StdInc.h"
 
+using StoreShadowToBeRendered_t = int(__cdecl*)(char type, RwTexture* texture, CVector* pos, float x1, float y1, float x2, float y2, __int16 intensity, char r,
+                                                char g, char b, float zDistance, char bDrawOnWater, float scale, void* shadowData, char bDrawOnBuildings);
+auto StoreShadowToBeRendered = (StoreShadowToBeRendered_t)0x707390;
+
+CFxSA::CFxSA(CFxSAInterface* pInterface) : m_pInterface(pInterface)
+{
+
+    RwTexture* pRwTexture;
+    auto shadowTypeCount = static_cast<std::underlying_type_t<eShadowType>>(eShadowType::_COUNT);
+    for (auto i = 0; i < shadowTypeCount; i++)
+    {
+        auto textureAddressPtr = (void**)(SHADOW_BASE_TEXTURE_OFFSET + i * 4);
+        void* textureAddress = *textureAddressPtr;
+        pRwTexture = reinterpret_cast<RwTexture*>(textureAddress);
+        m_textureMap[static_cast<eShadowType>(i)] = pRwTexture;
+    }
+    m_textureMap[eShadowType::PLANE] = new RwTexture();
+}
```

Comment:
I think this can crash the program. You don't allocate memory for Rw textures using the new operator. You have to use `RwTextureCreate`. I want to know where you are deleting this texture? It looks like this will cause a memory.

Reply:
I dont know how properly use wrcreatetexture, and current code doesnt crash